Falcon40B: Save benchmark results for Superset

pbaraTT opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/26533

### Problem description
We should be pushing Falcon40B performance metrics into a Superset dashboard ([link](https://superset.tenstorrent.com/superset/dashboard/5/?native_filters_key=3t0bDxa_IV0)) so we can easily triage regressions and track performance.

### What's changed
Added saving benchmark data in Falcon40B's demo script, similarly to those of Mamba and Falcon7B models. Whenever the demo is run in CI environment, it saves a PKL file with all measurements, to be used in Superset.
Followed some of the changes from a closed PR https://github.com/tenstorrent/tt-metal/pull/11548



### Checklist
- [x] [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes for Falcon40B with the two newly enabled steps
    https://github.com/tenstorrent/tt-metal/actions/runs/17273277670/job/49024523277
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes

pbaraTT committed:
Falcon40B: Save benchmarking data to json - first pass

pbaraTT committed:
Log two more metrics: prefill TTFT, prefill+decode t/s/u

skhorasganiTT commented:
The "save environment data" and "upload benchmark data" steps in the github workflow yaml for t3k demo tests (https://github.com/tenstorrent/tt-metal/blob/main/.github/workflows/t3000-demo-tests-impl.yaml#L92-L110) also need to be modified to include Falcon40 (i.e. an extra condition in the two if statements `if: ${{ (matrix.test-group.name == 't3k_falcon7b_tests' .....`).
Also please test this by running https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml on your branch (and post the run link in the PR description) and verifying that in the Falcon40 demo test you see a log similar to https://github.com/tenstorrent/tt-metal/actions/runs/17142166627/job/48631837948#step:9:974 and checking that the save environment data" and "upload benchmark data" steps pass.

pbaraTT committed:
add the rest of necessary profiler entries for saving benchmark data

pbaraTT committed:
t3000-demo-tests CI job: add Falcon40b to list of supported models for uploading bench data

- add "t3k_falcon40b_tests" test group to list of supported cases for the two steps:
  - Save environment data
  - Upload benchmark data

pbaraTT committed:
Add performance targets for Falcon40B

- Figures pulled from https://github.com/tenstorrent/tt-metal/tree/v0.59.0-rc38#llms

pbaraTT committed:
Falcon40B demo test: parametrize perf_mode

pbaraTT commented on models/demos/t3000/falcon40b/demo/demo.py:
```diff
@@ -182,41 +184,53 @@ def run_falcon_demo_kv(
 
+        N_warmup_iter = {"inference_prefill": 5, "inference_decode": 5}
+        perf_targets = {
+            "prefill_t/s": None,
```
This number is not provided in the table from [models README](https://github.com/tenstorrent/tt-metal/tree/main/models#llms), so I left it blank

skhorasganiTT replied:
For the other models we've been using the targets field in the Superset to track the aimed perf target (not the perf we're achieving today). Based on the LLM table (https://github.com/tenstorrent/tt-metal/tree/main/models#llms) for Falcon40 for decode it is 36 t/s/u (not 11.9 as you currently have). For prefill it can be left blank. 

CI perf targets are orthogonal and can be addressed later.

skhorasganiTT replied:
Thanks for doing the investigation @pbaraTT. Seems like an orthogonal issue so shouldn't block this PR, but please create a separate Git issue and link here.

pbaraTT committed:
Update Falcon40B decode performance targets to correct

- Decode target: 36 t/s/u
